scripts/migrate_project_mailing_lists.py

```diff
@@ -0,0 +1,111 @@
+from website.notifications.utils import to_subscription_key
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+EVENT = 'mailing_list_events'
+SUBSCRIPTION_TYPE = 'email_transactional'
+
+def fake(*args, **kwargs):
+    # Override actual method to prevent outgoing calls
+    return
+
+def get_targets():
+    return Node.find()
+
+def migrate(dry_run=True):
+    if dry_run:
+        create_list = fake
+    else:
+        create_list = real_create_list
+
+    successful_enables = []
+    successful_disables = []
+    unknown_failures = {}
+    nodes = get_targets()
+    ncount = len(nodes)
+    logger.info('Preparing to migrate {} nodes.'.format(ncount))
+    for i, node in enumerate(nodes):
+        if node.is_registration or node.is_collection or node.is_deleted:
+            try:
+                logger.info('({0}/{1})Disabling mailing list for registration/dashboard {2}'.format(i+1, ncount, node._id))
+                node.mailing_enabled = False    
+                node.save()
+                successful_disables.append(node._id)
+            except Exception as e:
+                unknown_failures[node._id] = e
+        else:
+            try:
+                subscription = NotificationSubscription(
+                    _id=to_subscription_key(node._id, EVENT),
+                    owner=node,
+                    event_name=EVENT
+                )
+                logger.info('({0}/{1})Enabling mailing list for node {2}'.format(i+1, ncount, node._id))
+                node.mailing_enabled = True
+
+                for user in node.contributors:
+                    if user.is_active:
+                        logger.info('Subscribing user {} on node {}'.format(user, node))
+                        subscription.add_user_to_subscription(user, SUBSCRIPTION_TYPE)
+                        # users added on `create_list`
+
+                subscription.save()
+                node.save()
+                successful_enables.append(node._id)
+            except Exception as e:
+                unknown_failures[node._id] = e
+            else:
+                create_list(node._id)
+        if i % 100 == 0:
+            for key in ('node', 'user', 'fileversion', 'storedfilenode'):
+                Node._cache.data.get(key, {}).clear()
+                Node._object_cache.data.get(key, {}).clear()
+
+
+    logger.info(
+        "Successfully enabled {0} new mailing lists for nodes:\n{1}".format(
+            len(successful_enables), successful_enables
+        )
+    )
+
+    logger.info(
+        "Successfully disabled {0} new mailing lists for registrations/dashboards:\n{1}".format(
+            len(successful_disables), successful_disables
+        )
+    )
+
+    if unknown_failures:
+        logger.error('Handled {0} unknown exceptions while creating lists:\n{1}'.format(
+                len(unknown_failures), ['({}, {})'.format(nid, unknown_failures[nid]) for nid in unknown_failures.keys()]
```

Comment:
Not terribly concerned -- this won't be hit on an ideal database. When this was last tested with prod data, there were two `unknown_failures` that turned out to be old `NodeFile` guids which were somehow grabbed.